cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/cache/chart_cache.go

```diff
@@ -329,6 +329,47 @@ func (c *ChartCache) DeleteChartsForRepo(repo *types.NamespacedName) error {
 	return nil
 }
 
+func (c *ChartCache) DeleteChartsForRepo(repo *types.NamespacedName) error {
+	log.Infof("+DeleteChartsForRepo(%s)", repo)
+	defer log.Infof("-DeleteChartsForRepo(%s)", repo)
+
+	return c.deleteChartsHelper(repo, sets.String{})
+}
+
+func (c *ChartCache) PurgeObsoleteChartVersions(keepThese []models.Chart) error {
+	log.Infof("+PurgeObsoleteChartVersions()")
+	defer log.Infof("-PurgeObsoleteChartVersions")
+
+	repos := map[types.NamespacedName]sets.String{}
+	for _, ch := range keepThese {
+		if ch.Repo == nil {
+			continue
```

Comment:
Should that be an error if this function is called with a chart with a nil repo? Or at least, logging the issue? Wondering why we'd not want to report that somehow?

Reply:
shouldn't ever happen, but you're right. Let me at least log it